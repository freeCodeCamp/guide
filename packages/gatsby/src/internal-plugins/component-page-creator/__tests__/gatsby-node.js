const validatePath = require(`../validate-path`)
const createPath = require(`../create-path`)

describe(`JavaScript page creator`, () => {
  it(`filters out files that start with underscores`, () => {
    const files = [
      {
        path: `something/blah.js`,
      },
      {
        path: `something/_blah.js`,
      },
      {
        path: `_blah.js`,
      },
    ]

    expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
  })

  it(`filters out files that start with template-*`, () => {
    const files = [
      {
        path: `something/blah.js`,
      },
      {
        path: `something/template-cool-page-type.js`,
      },
      {
        path: `template-cool-page-type.js`,
      },
    ]

    expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
  })

  it(`Creates paths`, () => {
    const basePath = `/a/`
    const paths = [`/a/b/c/de`, `/a/bee`, `/a/b/d/c/`]

    expect(paths.map(p => createPath(basePath, p))).toMatchSnapshot()
  })
})
