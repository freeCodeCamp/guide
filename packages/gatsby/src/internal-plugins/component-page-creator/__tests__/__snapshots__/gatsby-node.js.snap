// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript page creator Creates paths 1`] = `
Array [
  "/b/c/de/",
  "/bee/",
  "/b/d/c/",
]
`;
